File: regexlib/javascript_test_file/regexlib_8266.js
/* 8266
 * ^(?i:([a-z0-9!#$%^&*{}'`+=-_|/?]+(?:\.[a-z0-9!#$%^&*{}'`+=-_|/?]+)*)@([a-z0-9]+\z?.*[a-z0-9-_]+)*(\.[a-z0-9]{2,}))$
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:"ia@"+"a"*16+"! _1SLQ_2"
 */
var REGEX = new RegExp("^(?i:([a-z0-9!#$%^&*{}'`+=-_|/?]+(?:\.[a-z0-9!#$%^&*{}'`+=-_|/?]+)*)@([a-z0-9]+\z?.*[a-z0-9-_]+)*(\.[a-z0-9]{2,}))$")
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = 'ia@' + 'a'.repeat(i*1) + '! _1SLQ_2'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}

File: regexlib/python_re_test_file/regexlib_8266.py
# 8266
# ^(?i:([a-z0-9!#$%^&*{}'`+=-_|/?]+(?:\.[a-z0-9!#$%^&*{}'`+=-_|/?]+)*)@([a-z0-9]+\z?.*[a-z0-9-_]+)*(\.[a-z0-9]{2,}))$
# EXPONENT
# nums:4
# EXPONENT AttackString:"ia@"+"a"*16+"! _1SLQ_2"

import re
from time import perf_counter

regex = """^(?i:([a-z0-9!#$%^&*{}'`+=-_|/?]+(?:\.[a-z0-9!#$%^&*{}'`+=-_|/?]+)*)@([a-z0-9]+\z?.*[a-z0-9-_]+)*(\.[a-z0-9]{2,}))$"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "ia@" + "a" * i * 1 + "! _1SLQ_2"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")